### PR TITLE
Azure ci

### PR DESCRIPTION
**Pull request by VLitanADI**
Implements issue #483

### CODE REVIEW

**Comment by VLitanADI** [comment]
Should I squash all commits?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      eq(variables.CACHE_RESTORED, 'false'), 
+      eq(variables['BUILD_TYPE'], 'clang_format'),
+      eq(variables['BUILD_TYPE'], 'cppcheck'),
+      eq(variables['BUILD_TYPE'], 'deploy_doxygen')
+      )
+      
+    displayName: "Install Dependencies"
+
+  - script: ./ci/azure/run_build.sh
+    displayName: "Build"
```
Comment (DanielGuramulta) [inline azure-pipelines.yml:139]: no newline at the end of this file

File: ci/azure/build_docker_image.sh [hunk 2]
```diff
@@ -2,16 +2,16 @@
 
 # Build the given dockerfile with the given name
 
-. ci/travis/lib.sh
+. ci/azure/lib.sh
 
 mkdir -p temp_deps
 
-export CMAKE_OPTIONS="-DWITH_OPENCV=on"
+# export CMAKE_OPTIONS="-DWITH_OPENCV=on"
```
Comment (DanielGuramulta) [inline ci/azure/build_docker_image.sh:9]: if this is not needed, lets remove it

File: ci/azure/setup_paths.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -ex
+
+#we need to export all those variables from a script because of pwd
+setup_absolute_paths() {
+    export WORK_DIR=$(pwd) 
+    export BUILD_DIR="${WORK_DIR}/build"
+    export DEPS_DIR="${BUILD_DIR}/deps"
+}
+
+setup_absolute_paths
```
Comment (DanielGuramulta) [inline ci/azure/setup_paths.sh:12]: missing new line

**Review by DanielGuramulta — APPROVED** [review]
Looks good. I think squashing the commits is a good idea!

**Pull request merged**